src/main/java/carparks/parking/ParkingService.java

```diff
@@ -0,0 +1,190 @@
+package carparks.parking;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ParkingService {
+
+	private ParkingDao parkingDao;
+	private FeeCalculator feeCalculator;
+	private Optional<List<CurrencyConverter>> converters;
+
+	@Autowired(required=false)
+	public ParkingService(ParkingDao parkingDao, FeeCalculator feeCalculator, Optional<List<CurrencyConverter>> converters) {
+		this.parkingDao = parkingDao;
+		this.feeCalculator = feeCalculator;
+		this.converters = converters;
+	}
+
+	public ParkingEntryDto tryToCreateParkingEntry(String plateNumber, FeeType feeType) {
```

Comment:
Jak by się zachowała ta metoda jakby pomiędzy po wejściu do bloku `else`, inny wątek wywołał `getStartedSessionsListByPlateNumber`? Wiesz w jaki sposób można rozwiązać ten problem?

Reply:
Z wielowątkowością pracowałem tylko w C, więc wiem, że niskopoziomowo potrzebujemy jakiegoś mutexa, który będzie blokował używane entries na czas wykonywania if'a w metodzie tryToCreateParkingEntry. W Javie, pewnie da się to zrobić na trochę wyższym poziomie abstrakcji - na pierwszy rzut oka keyword "synchronized" zdaje się załatwiać sprawę, ale musiałbym poczytać jeszcze o tym.

Reply:
A czy wiesz, czy rozwiązanie z mutexem/synchronized sprawdziłoby się jakbyś miał uruchomioną usługę na dwóch instancjach?

Reply:
Wydaje mi się, że nie, bo zasoby nie byłyby współdzielone. Chyba, że można jakoś współdzielić zasoby między dwoma JVMami? Ale to tylko intuicja i rozważania, nie wiem nawet czy tak można.

Reply:
Można wykorzystać do tego celu bazę - wystarczyłby odpowiedni constraint ale w bardziej skomplikowanych przypadkach trzeba robić optimistic/pessimistic locking

Reply:
Ok, czyli rozumiem, że wtedy trzeba to jakoś zrobić na poziomie transakcji na bazie, czyli faktycznie mutex/synchronized nie zadziała?